pkg/controller/jobs/leaderworkerset/leaderworkerset_webhook.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package leaderworkerset
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apivalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	leaderworkersetv1 "sigs.k8s.io/lws/api/leaderworkerset/v1"
+
+	"sigs.k8s.io/kueue/pkg/controller/constants"
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	podcontroller "sigs.k8s.io/kueue/pkg/controller/jobs/pod"
+	"sigs.k8s.io/kueue/pkg/queue"
+	utilpod "sigs.k8s.io/kueue/pkg/util/pod"
+)
+
+type Webhook struct {
+	client                       client.Client
+	manageJobsWithoutQueueName   bool
+	managedJobsNamespaceSelector labels.Selector
+	queues                       *queue.Manager
+}
+
+func SetupWebhook(mgr ctrl.Manager, _ ...jobframework.Option) error {
+	wh := &Webhook{
+		client: mgr.GetClient(),
+	}
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&leaderworkersetv1.LeaderWorkerSet{}).
+		WithDefaulter(wh).
+		WithValidator(wh).
+		Complete()
+}
+
+// +kubebuilder:webhook:path=/mutate-leaderworkerset-x-k8s-io-v1-leaderworkerset,mutating=true,failurePolicy=fail,sideEffects=None,groups="leaderworkerset.x-k8s.io",resources=leaderworkersets,verbs=create;update,versions=v1,name=mleaderworkerset.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomDefaulter = &Webhook{}
+
+func (wh *Webhook) Default(ctx context.Context, obj runtime.Object) error {
+	lws := fromObject(obj)
+	log := ctrl.LoggerFrom(ctx).WithName("leaderworkerset-webhook")
+	log.V(5).Info("Applying defaults")
+
+	jobframework.ApplyDefaultLocalQueue(lws.Object(), wh.queues.DefaultLocalQueueExist)
+	suspend, err := jobframework.WorkloadShouldBeSuspended(ctx, lws.Object(), wh.client, wh.manageJobsWithoutQueueName, wh.managedJobsNamespaceSelector)
+	if err != nil || !suspend {
```

Comment:
nit, I would prefer to keep the code consistent with STS https://github.com/kubernetes-sigs/kueue/blob/b7a111188887ade2b727cd56b1a2edc9fcceddd0/pkg/controller/jobs/statefulset/statefulset_webhook.go#L73-L77 for consistency and readability.

Reply:
Done